lms/djangoapps/experiments/utils.py

```diff
@@ -73,6 +74,22 @@
 )
 # TODO END: Clean up as part of REV-1205 (End)
 
+# .. toggle_name: streak_celebration.AA-759
+# .. toggle_implementation: ExperimentWaffleFlag
+# .. toggle_default: False
+# .. toggle_description: This experiment flag enables an engagement discount incentive message.
+# .. toggle_warnings: This flag depends on the streak celebration feature being enabled
+# .. toggle_use_cases: temporary
+# .. toggle_creation_date: 2021-05-05
+# .. toggle_target_removal_date: 2021-07-05
+# .. toggle_tickets: https://openedx.atlassian.net/browse/AA-759
+AA759_FLAG = ExperimentWaffleFlag(
+    LegacyWaffleFlagNamespace(name='streak_celebration'),
+    'AA-759',
```

Comment:
I don't love that this experiment flag name isn't descriptive at all. Someone looking at the django admin page wouldn't have much context. Maybe `streak_celebration.discount_message` or something? When we implement new settings, we don't do `ENABLE_AA304`, it's just `ENABLE_PUBLISHER` or similar. I take the point that this isn't really for public consumption and is short lived so it doesn't have to look nice, but waffles tend to stick around longer than expected.

Especially for the variable name -- the code would be much more readable if it was `STREAK_DISCOUNT_FLAG.is_enabled()` vs `AA759_FLAG.is_enabled()`, in my opinion.

If you still prefer using a ticket name as the waffle name, at least write a nice note in the django admin page for this flag with a link to the ticket. (I mean, that's probably good regardless.) Or maybe use a human readable name _plus_ the ticket. But I really think the python name shouldn't just be the ticket.

Reply:
sure I'm happy to use both the ticket name and the human readable name
how about streak_celebration.discount_experiment_AA759?

personally I'm a fan of using the ticket where possible - if I'm not someone who specifically worked on this experiment I'll have no idea what it is talking about. Plus, human readable names are always used inconsistently

for example, I was looking for the template that Christa created for this experiment which she called Audit Learner Engage Test (which didn't have the words streak or discount) so I was a bit confused, just being able to look for the ticket would have been nice, or to go directly to jira to look up the context if I came across the template

or if we take the flag effort_estimation.location, if I'm not familiar with the project and I come across the code and I want to learn more about what it is, being able to go directly to the ticket would be quite valuable IMO
of course, once something becomes a real feature and not an experiment, then I think a ticket is less useful because at that point hopefully the feature has been documented somewhere more permanent